### PR TITLE
support using zstd decompress checkpoint

### PR DESCRIPTION
**Pull request by xyyy1420**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,6 +28,7 @@ remove_quote = $(patsubst "%",%,$(1))
 
 ISA    ?= $(if $(CONFIG_ISA),$(call remove_quote,$(CONFIG_ISA)),x86)
 CFLAGS += -D__ISA__=$(ISA)
+CFLAGS += -lzstd
```
Comment (poemonsense) [inline Makefile:31]: Only include the header when CONFIG_MEM_COMPRESS is enabled?

File: Makefile [hunk 2]
```diff
@@ -28,6 +28,7 @@ remove_quote = $(patsubst "%",%,$(1))
 
 ISA    ?= $(if $(CONFIG_ISA),$(call remove_quote,$(CONFIG_ISA)),x86)
 CFLAGS += -D__ISA__=$(ISA)
+
```
Comment (poemonsense) [inline Makefile:31]: Unused empty line

**Review by poemonsense — COMMENTED** [review]
LGTM. Let zyy have a look

File: src/monitor/image_loader.c [hunk 3]
```diff
@@ -63,6 +65,131 @@ long load_gz_img(const char *filename) {
...
+      free(decompress_file_buffer);
+
+      return -1;
+    }
+
+    if (output.pos == 0) {
+      break;
+    }
+
+    for (uint32_t x = 0; x < output.pos; x++) {
```
Comment (shinezyy) [inline src/monitor/image_loader.c:165]: Can we use uint64_t here? U64 might be faster

**Review by shinezyy — CHANGES_REQUESTED** [review]

**Review by shinezyy — APPROVED** [review]

**Pull request merged**